ui/article-renderer/src/main/java/org/cru/godtools/article/fragment/ArticlesFragment.java

```diff
@@ -124,6 +138,47 @@ private void setupViewModel() {
         }
     }
 
+    // region refresh Layout
+    private void setUpSwipeRefresh() {
+        if (mSwipeRefreshLayout == null) {
+            return;
+        }
+
+        mSwipeRefreshLayout.setOnRefreshListener(this);
+    }
+
+    private void setRefreshing(boolean isRefreshing) {
+        if (mSwipeRefreshLayout != null) {
+            mSwipeRefreshLayout.setRefreshing(isRefreshing);
+        }
+    }
+
+    @Override
+    public void onRefresh() {
+        Timber.d("onRefresh() called");
+        setUpListenableFuture();
+    }
+
+    private void setUpListenableFuture() {
+        AEMDownloadManger manger = AEMDownloadManger.getInstance(requireContext());
+        ListenableFuture<List<Boolean>> future = manger.enqueueSyncManifestAemImports(mManifest, true);
+        Futures.addCallback(future, new FutureCallback<List<Boolean>>() {
+            @Override
+            public void onSuccess(List<Boolean> result) {
+                setRefreshing(false);
+            }
+
+            @Override
+            public void onFailure(Throwable t) {
+                Timber.e(t);
+                setRefreshing(false);
+            }
+        }, new MainThreadExecutor());
```

Comment:
Because you don't really care whether the future succeeds or fails, it looks like `ListenableFuture.addListener()` will allow you to simplify this to:
```
        future.addListener(() -> setRefreshing(false), new MainThreadExecutor());
```

Reply:
Ok I will try that.

Reply:
the next thing to do is address the potential memory leak by creating a static inner class Runnable with a weak reference to the `SwipeToRefreshLayout`. that way if the view goes away before the sync completes the listener doesn't hold a hard reference to the fragment.

Reply:
I might have something that already does this, let me take a quick look.

Reply:
nope, don't have anything that does that immediately. although maybe a `WeakRunnable` class could make sense.